crates/ruff_server/src/server/api/requests/format.rs

```diff
@@ -64,7 +64,8 @@ pub(super) fn format_document(snapshot: &DocumentSnapshot) -> Result<super::Form
     let text_document = snapshot
         .query()
         .as_single_document()
-        .expect("format should only be called on text documents or notebook cells");
+        .map_err(|err| anyhow::anyhow!("Failed to get text document for the format request: {err}"))
+        .unwrap();
```

Comment:
I never fully understood this in the LSP but could we propagate the error instead of unwrapping here?

```suggestion
        .context("Failed to get text document for the format request"))
        .unwrap();
```

I think that should be sufficient. `anyhow` preserves the entire error-chain

Reply:
Yeah, I thought about doing this but then we would still have the problem that the message will only be logged if the user has turned on server messages via `ruff.trace.server: "messages"`:

https://github.com/astral-sh/ruff/blob/4ce0c42dac09039e86311fdb2da5558947e18188/crates/ruff_server/src/server/api.rs#L197-L214

Reply:
That makes sense. I would still prefer propagating because there's a workaround to get logs, but there's no workaround for a crashing ruff server.

Reply:
My main concern here is that the user will only see the notification but no log messages. Then, the user would need to turn on server messages which would restart the server and will loose the state in which the panic occurs. So, the user would need to wait until the next time the error occurs to provide us the logs.

Reply:
Using anyhow's `Context` requires cloning the `Url` because I can't use the reference because of lifetime issues.

Reply:
Can you use `with_context`

Reply:
Oh I see now. I think cloning is fine. It only happens in the error case and it's uncommon that errors have lifetime bounds (e.g. it prevents users from propagating errors, as you've seen here)